src/toil/wdl/wdl_analysis.py

```diff
@@ -620,18 +655,35 @@ def parse_declaration_type(self, typeAST):
         which Toil needs to import, so we recursively travel down to the innermost
         type which will tell us if the variables are files that need importing.
 
+        For Pair and Map compound types, we recursively travel down the subtypes and
```

Comment:
Shouldn't we use this same recursive traversal for `Array`? And optional types would then want to be, internally, `Optional[subtype]`, which just gets parsed as a `?` suffix.